Sources/MongoSwift/MongoCollection+BulkWrite.swift

```diff
@@ -325,6 +325,18 @@ public struct BulkWriteOptions: Codable {
         self.ordered = options.ordered
         self.writeConcern = options.writeConcern
     }
+
+    private enum CodingKeys: String, CodingKey {
+        case bypassDocumentValidation, ordered, writeConcern
+    }
+
+    // A manual implementation is required to enforce a default value for ordered without making it optional.
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.bypassDocumentValidation = try container.decodeIfPresent(Bool.self, forKey: .bypassDocumentValidation)
+        self.ordered = try container.decodeIfPresent(Bool.self, forKey: .ordered) ?? true
```

Comment:
if we provide a default value like `public var ordered = true` does the synthesized implementation not use that when ordered is missing?

Reply:
I had the same thought. unfortunately no, that default value does not get propagated to the Codable conformance.